deps/rabbit/src/rabbit_node_monitor.erl

```diff
@@ -323,20 +326,53 @@ find_blocked_global_peers1([], _) ->
 
 unblock_global_peer(PeerNode) ->
     ThisNode = node(),
-    PeerState = rpc:call(PeerNode, sys, get_status, [global_name_server]),
+
+    {status, _, _, [_, _, _, _, PeerState]} = PeerStatus =
+        erpc:call(PeerNode, sys, get_status, [global_name_server]),
+    {status, _, _, [_, _, _, _, ThisState]} = ThisStatus =
+        sys:get_status(global_name_server),
+
+    PeerToThisCid = connection_id(PeerState, ThisNode),
+    ThisToPeerCid = connection_id(ThisState, PeerNode),
+
+    logger:info(
+      "global hang workaround: faking nodedown / nodeup between peer node ~s "
+      "(connection ID to us: ~p) and our node ~s (connection ID to peer: ~p)",
+      [PeerNode, PeerToThisCid, ThisNode, ThisToPeerCid]),
     logger:debug(
-      "Global hang workaround: global state on ~s seems inconsistent~n"
-      " * Peer global state:  ~p~n"
-      " * Local global state: ~p~n"
-      "Faking nodedown/nodeup between ~s and ~s",
-      [PeerNode, PeerState, sys:get_status(global_name_server),
-       PeerNode, ThisNode]),
-    {global_name_server, ThisNode} ! {nodedown, PeerNode},
-    {global_name_server, PeerNode} ! {nodedown, ThisNode},
-    {global_name_server, ThisNode} ! {nodeup, PeerNode},
-    {global_name_server, PeerNode} ! {nodeup, ThisNode},
+      "peer global state: ~p~nour global state: ~p",
+      [PeerStatus, ThisStatus]),
+
+    {ThisDownMsg, ThisUpMsg} = messages(ThisToPeerCid, PeerNode),
+    {PeerDownMsg, PeerUpMsg} = messages(PeerToThisCid, ThisNode),
+    {global_name_server, ThisNode} ! ThisDownMsg,
+    {global_name_server, PeerNode} ! PeerDownMsg,
+    {global_name_server, ThisNode} ! ThisUpMsg,
+    {global_name_server, PeerNode} ! PeerUpMsg,
     ok.
 
+connection_id(Misc, Node) ->
+    [State] = [S || {data, [{"State", S}]} <- Misc],
+    state = element(1, State),
```

Comment:
Could you please use `?assertEqual(state, element(1, State)),` instead? It will raise an exception which will make it clear we hit an assertion.

You will have to `-include_lib("stdlib/include/assert.hrl").`, not the EUnit header.

Reply:
Thanks, that's a good tip. Given that I changed back to `sys:get_state/1`, I completely removed that assertion. (The assertion was mainly meant to check that we parse the state correctly from `sys:get_status/1`.)